app/scripts/modules/core/src/managed/ManagedResourceHistoryModal.tsx

```diff
@@ -0,0 +1,227 @@
+import React from 'react';
+import classNames from 'classnames';
+import { DateTime } from 'luxon';
+import { UISref } from '@uirouter/react';
+
+import {
+  ModalHeader,
+  ModalBody,
+  Table,
+  TableRow,
+  TableCell,
+  standardGridTableLayout,
+  useData,
+} from 'core/presentation';
+
+import { relativeTime, timestamp } from 'core/utils';
+import { IManagedResourceSummary, IManagedResourceDiff } from 'core/domain';
+import { AccountTag } from 'core/account';
+import { ManagedReader } from 'core/managed';
+import { Spinner } from 'core/widgets';
+
+import { ManagedResourceDiffTable } from './ManagedResourceDiffTable';
+
+import './ManagedResourceHistoryModal.less';
+
+export interface IManagedResourceHistoryModalProps {
+  resourceSummary: IManagedResourceSummary;
+}
+
+const { useMemo } = React;
+
+const viewConfigurationByEventType = {
```

Comment:
@gcomstock fyi, I commented in here some event types that we don't currently have bespoke icons for. In the absence of specific icons, I've temporarily chosen some existing MD icons that I think are close enough to work for now (you can see some of them in the screenshots)